### PR TITLE
LFVM: Check jump instructions JUMPDEST test

### PR DESCRIPTION
**Pull request by LuisPH3**
part of #751 

This test verifies that every jump operation does JUMPDEST check.

### CODE REVIEW

File: go/interpreter/lfvm/instructions_test.go [hunk 1]
```diff
@@ -1173,3 +1173,71 @@ func TestOpEndWithResult_ReportOverflow(t *testing.T) {
...
+
+			ctxt := context{
+				params: tosca.Parameters{
+					BlockParameters: tosca.BlockParameters{
+						Revision: tosca.R13_Cancun,
+					},
+				},
+				stack:  NewStack(),
+				memory: NewMemory(),
+				gas:    1 << 32,
```
Comment (facuMH) [inline go/interpreter/lfvm/instructions_test.go:1230]: unnecessary specifications.

**Review by facuMH — APPROVED** [review]

File: go/interpreter/lfvm/instructions_test.go (hunk 1 above)
Comment (LuisPH3) [reply go/interpreter/lfvm/instructions_test.go:1230]: gas is needed, but memory not.

Comment (facuMH) [reply go/interpreter/lfvm/instructions_test.go]: sorry, yes, gas does not need to be so big, but it's fine as is

**Review by facuMH — DISMISSED** [review]

File: go/interpreter/lfvm/instructions_test.go [hunk 2]
```diff
@@ -1291,3 +1291,61 @@ func TestInstructions_EIP2929_SSTOREReportsOutOfGas(t *testing.T) {
 		}
 	}
 }
+
+func TestInstructions_JumpInstructionsCheckJUMPDEST(t *testing.T) {
```
Comment (HerbertJordan) [inline go/interpreter/lfvm/instructions_test.go:1295]: For a follow-up: a related test that should be added for conditional jumps: if the condition is false, it must not test the jump target for pointing to a JUMPDEST; invalid target destinations must be accepted if the branch is not taken.

**Review by HerbertJordan — APPROVED** [review]
Nice, thanks!

File: go/interpreter/lfvm/instructions_test.go (hunk 2 above)
Comment (LuisPH3) [reply go/interpreter/lfvm/instructions_test.go]: I think this is not that far off. I extend the scope of this PR.

**Review by facuMH — APPROVED** [review]

**Pull request merged**